catalog/dags/providers/provider_api_scripts/phylopic.py

```diff
@@ -35,15 +39,43 @@ def __init__(self, *args, **kwargs):
 
     def ingest_records(self):
         self._get_initial_query_params()
-        super().ingest_records()
+        try:
+            super().ingest_records()
+
+        # Catch 410 error caused by the build_param changing while ingestion is ongoing
+        except HTTPError as error:
+            if error.response.status_code == 410:
+                # Refetch initial query params; this will update the build_param to the
+                # most recent value and reset the `current_page` to 1.
```

Comment:
Could this cause an infinite loop when the `build_param` keeps changing in the middle of the ingestion, and we keep resetting the page to 1? Do we need a mechanism to prevent that?

Reply:
Good question -- it should not! Only the first call to `super().ingest_records()` is in the `try`; if it raises an error again on the _second_ attempt, it will not be caught. I though it was fair to only retry once, to prevent the infinite loop as you mentioned.

You can verify this using the same code from the testing instructions, but in `get_initial_query_params` add:

```
+        if self.test == 0:
+            self.build_param = 307
+        if self.test == 1:
+            self.build_param = 218
+        self.test += 1
```

(You need to change the build_param to a different, also incorrect build param on the second run). The DAG errors after both bad build params have been tried.

Reply:
Retrying only once makes sense (albeit I don't fully understand what `build_param` is and how and why it can change during ingestion :) ).

@stacimc, I think this PR also fixes this issue: [Bubble up original exception when retries have exceeded](https://github.com/WordPress/openverse/issues/1369)